### PR TITLE
时间序列分析、可视化、和使用 LSTM 预测

### LINKED ISSUE #5824: 时间序列分析、可视化、和使用 LSTM 预测
* 原文链接：[Time Series Analysis, Visualization & Forecasting with LSTM](https://towardsdatascience.com/time-series-analysis-visualization-forecasting-with-lstm-77a905180eba)
* Markdown文件：[文件地址](https://github.com/xitu/gold-miner/blob/master/TODO1/time-series-analysis-visualization-forecasting-with-lstm.md)
* PR 地址：https://github.com/xitu/gold-miner/pull/5848
* 文章分类：**AI**
* **注意：文件位置在 `TODO1` 文件夹中，不是之前的 `TODO` 了。**
----
* 翻译时间：4 天
* 校对时间：1 天
* 翻译积分：3.5-4.5 分
* 校对积分：1-2 分
* [积分有什么用](https://github.com/xitu/gold-miner/wiki)

### PR DESCRIPTION
**Pull request by Minghao23**
译文翻译完成，resolve #5824

### CODE REVIEW

**Comment by Xuyuey** [comment]
校对认领

**Comment by fanyijihua** [comment]
@Xuyuey 好的呢 :beer:

File: TODO1/time-series-analysis-visualization-forecasting-with-lstm.md [hunk 1]
```diff
@@ -2,24 +2,24 @@
...
+标题已经阐述了一切。
 
-Without further ado, let’s roll!
+闲话少说，让我们直接开始吧！
 
-## The Data
+## 数据
 
-The data is the measurements of electric power consumption in one household with a one-minute sampling rate over a period of almost 4 years that can be downloaded from [here](https://www.kaggle.com/uciml/electric-power-consumption-data-set).
+该数据是对一个家庭在近四年的时间里以一分钟采样率测量的电力消耗，可以在[这里](https://www.kaggle.com/uciml/electric-power-consumption-data-set)进行下载。
```
Comment (Xuyuey) [inline TODO1/time-series-analysis-visualization-forecasting-with-lstm.md]:
```suggestion
该数据是在近四年的时间里对一个家庭以一分钟采样率测量的电力消耗，可以在[这里](https://www.kaggle.com/uciml/electric-power-consumption-data-set)下载。
```

File: TODO1/time-series-analysis-visualization-forecasting-with-lstm.md [hunk 2]
```diff
@@ -230,13 +230,13 @@ ax5.tick_params(axis='both', which='major');
 
 ![](https://cdn-images-1.medium.com/max/2942/1*m70T6psxaS64c_GWt7uj-w.png)
 
-![Figure 5](https://cdn-images-1.medium.com/max/2930/1*CFEGvZ4t-iPKZH8ciOjEew.png)
+![图 5](https://cdn-images-1.medium.com/max/2930/1*CFEGvZ4t-iPKZH8ciOjEew.png)
 
-In general, our time series does not have a upward or downward trend. The highest average power consumption seems to be prior to 2007, actually it was because we had only December data in 2007 and that month was a high consumption month. In another word, if we compare year by year, it has been steady.
+通常来说，我们的时间序列不会存在上升或下降的趋势。最高的平均耗电量似乎是在 2007 年之前，实际上这是因为我们在 2007 年只有 12 月的数据，而那个月是用电高峰月。也就是说，如果我们逐年比较，这个序列其实较为平稳。
```
Comment (Xuyuey) [inline TODO1/time-series-analysis-visualization-forecasting-with-lstm.md]: 这里或许说的是 2016 年只有 12  月的数据？你觉得呢

File: TODO1/time-series-analysis-visualization-forecasting-with-lstm.md [hunk 3]
```diff
@@ -259,83 +259,83 @@ plt.xlabel('')
...
+在统计学中，[**Dickey–Fuller test**](https://en.wikipedia.org/wiki/Dickey%E2%80%93Fuller_test) 检验了一个零假设，即单位根存在于自回归模型中。备择假设依据使用的检验方法的不同而不同，但是通常为[平稳性](https://en.wikipedia.org/wiki/Stationary_process)或[趋势平稳性](https://en.wikipedia.org/wiki/Trend_stationary)。
 
-Stationary series has constant mean and variance over time. Rolling average and the rolling standard deviation of time series do not change over time.
+平稳序列的均值和方差一直是常数。时间序列在滑动窗口下的均值和标准差不随时间变化。
 
-## Dickey-Fuller test
+## Dickey-Fuller 检验
 
-[**Null Hypothesis**](https://en.wikipedia.org/wiki/Null_hypothesis) (H0): It suggests the time series has a unit root, meaning it is non-stationary. It has some time dependent structure.
+[**零检验**](https://en.wikipedia.org/wiki/Null_hypothesis)（H0）：表明时间序列有一个单位根，意味着它是非平稳的。它包含一些时间相关的成分。
```
Comment (Xuyuey) [inline TODO1/time-series-analysis-visualization-forecasting-with-lstm.md]:
```suggestion
[**零检验**](https://en.wikipedia.org/wiki/Null_hypothesis)（H0）：表明时间序列有一个单位根，意味着它是非平稳的。它包含一些和时间相关的成分。
```

File: TODO1/time-series-analysis-visualization-forecasting-with-lstm.md [hunk 4]
```diff
@@ -259,83 +259,83 @@ plt.xlabel('')
...
 
-[**Alternate Hypothesis**](https://en.wikipedia.org/wiki/Alternative_hypothesis) (H1): It suggests the time series does not have a unit root, meaning it is stationary. It does not have time-dependent structure.
+[**备择检验**](https://en.wikipedia.org/wiki/Alternative_hypothesis)（H1）：表明时间序列不存在单位根，意味着它是平稳的。它不包含时间相关的成分。
```
Comment (Xuyuey) [inline TODO1/time-series-analysis-visualization-forecasting-with-lstm.md]:
```suggestion
[**备择检验**](https://en.wikipedia.org/wiki/Alternative_hypothesis)（H1）：表明时间序列不存在单位根，意味着它是平稳的。它不包含和时间相关的成分。
```

File: TODO1/time-series-analysis-visualization-forecasting-with-lstm.md [hunk 5]
```diff
@@ -401,27 +401,27 @@ X_train = np.reshape(X_train, (X_train.shape[0], 1, X_train.shape[1]))
...
 ```
 
-## Model Architecture
+## 模型结构
 
-* Define the LSTM with 100 neurons in the first hidden layer and 1 neuron in the output layer for predicting Global_active_power. The input shape will be 1 time step with 30 features.
-* Dropout 20%.
-* Use the MSE loss function and the efficient Adam version of stochastic gradient descent.
-* The model will be fit for 20 training epochs with a batch size of 70.
+* 定义 LSTM 模型，第一个隐藏层含有 100 个神经元，输出层含有 1 个神经元，用于预测 Global_active_power。输入的维度是包含 30 个特征的一个时间步长。
```
Comment (Xuyuey) [inline TODO1/time-series-analysis-visualization-forecasting-with-lstm.md]:
```suggestion
* 定义 LSTM 模型，第一个隐藏层含有 100 个神经元，输出层含有 1 个神经元，用于预测 Global_active_power。输入的维度是一个包含 30 个特征的时间步长。
```

File: TODO1/time-series-analysis-visualization-forecasting-with-lstm.md (hunk 2 above)
Comment (Minghao23) [reply TODO1/time-series-analysis-visualization-forecasting-with-lstm.md]: 看图片和之前数据集的描述的话应该是2006年的12月，这个原文好像是写错了，我加个译者注吧。

**Comment by TrWestdoor** [comment]
校对认领

**Comment by fanyijihua** [comment]
@TrWestdoor 妥妥哒 :beers:

File: TODO1/time-series-analysis-visualization-forecasting-with-lstm.md [hunk 6]
```diff
@@ -259,83 +259,83 @@ plt.xlabel('')
...
+平稳序列的均值和方差一直是常数。时间序列在滑动窗口下的均值和标准差不随时间变化。
 
-## Dickey-Fuller test
+## Dickey-Fuller 检验
 
-[**Null Hypothesis**](https://en.wikipedia.org/wiki/Null_hypothesis) (H0): It suggests the time series has a unit root, meaning it is non-stationary. It has some time dependent structure.
+[**零检验**](https://en.wikipedia.org/wiki/Null_hypothesis)（H0）：表明时间序列有一个单位根，意味着它是非平稳的。它包含一些和时间相关的成分。
 
-[**Alternate Hypothesis**](https://en.wikipedia.org/wiki/Alternative_hypothesis) (H1): It suggests the time series does not have a unit root, meaning it is stationary. It does not have time-dependent structure.
+[**备择检验**](https://en.wikipedia.org/wiki/Alternative_hypothesis)（H1）：表明时间序列不存在单位根，意味着它是平稳的。它不包含和时间相关的成分。
```
Comment (TrWestdoor) [inline TODO1/time-series-analysis-visualization-forecasting-with-lstm.md]:
```suggestion
[**备择检验**](https://en.wikipedia.org/wiki/Alternative_hypothesis)（H1）：表明时间序列不存在单位根，即它是平稳的。它不包含和时间相关的成分。
```
此处“表明”和“意味着”连着使用，显得有些累赘。

File: TODO1/time-series-analysis-visualization-forecasting-with-lstm.md [hunk 7]
```diff
@@ -355,26 +355,26 @@ def test_stationarity(timeseries):
 test_stationarity(df2.Global_active_power.dropna())
 ```
 
-![Figure 10](https://cdn-images-1.medium.com/max/2334/1*teCNF7iDamLWXTVFWOF3ew.png)
+![图 10](https://cdn-images-1.medium.com/max/2334/1*teCNF7iDamLWXTVFWOF3ew.png)
 
-From the above results, we will reject the null hypothesis H0, the data does not have a unit root and is stationary.
+从以上结论可得，我们会拒绝零检验 H0，数据没有单位根且是平稳的。
```
Comment (TrWestdoor) [inline TODO1/time-series-analysis-visualization-forecasting-with-lstm.md]:
```suggestion
从以上结论可得，我们会拒绝零检验 H0，因为数据没有单位根且是平稳的。
```
此处直译显示不出因果关系。

File: TODO1/time-series-analysis-visualization-forecasting-with-lstm.md [hunk 8]
```diff
@@ -401,27 +401,27 @@ X_train = np.reshape(X_train, (X_train.shape[0], 1, X_train.shape[1]))
 X_test = np.reshape(X_test, (X_test.shape[0], 1, X_test.shape[1]))
 ```
```
Comment (TrWestdoor) [inline TODO1/time-series-analysis-visualization-forecasting-with-lstm.md]: 注释也应该翻译。

File: TODO1/time-series-analysis-visualization-forecasting-with-lstm.md [hunk 9]
```diff
@@ -401,27 +401,27 @@ X_train = np.reshape(X_train, (X_train.shape[0], 1, X_train.shape[1]))
...
 
-## Model Architecture
+## 模型结构
 
-* Define the LSTM with 100 neurons in the first hidden layer and 1 neuron in the output layer for predicting Global_active_power. The input shape will be 1 time step with 30 features.
-* Dropout 20%.
-* Use the MSE loss function and the efficient Adam version of stochastic gradient descent.
-* The model will be fit for 20 training epochs with a batch size of 70.
+* 定义 LSTM 模型，第一个隐藏层含有 100 个神经元，输出层含有 1 个神经元，用于预测 Global_active_power。输入的维度是一个包含 30 个特征的时间步长。
+* 随机失活 20%。
```
Comment (TrWestdoor) [inline TODO1/time-series-analysis-visualization-forecasting-with-lstm.md]: 这个我个人觉得就 dropout 更好吧，很少见到有文章翻译这个词。

File: TODO1/time-series-analysis-visualization-forecasting-with-lstm.md [hunk 10]
```diff
@@ -440,7 +440,7 @@ print('Test Root Mean Squared Error:',np.sqrt(mean_squared_error(Y_test[0], test
```
Comment (TrWestdoor) [inline TODO1/time-series-analysis-visualization-forecasting-with-lstm.md]: 上面代码中也有一句注释没有翻译。

File: TODO1/time-series-analysis-visualization-forecasting-with-lstm.md [hunk 11]
```diff
@@ -476,13 +476,13 @@ plt.legend(fontsize=15)
 plt.show();
 ```
 
-![Figure 12](https://cdn-images-1.medium.com/max/2000/1*fNpJ18MfE32UD0ib3LnrfA.png)
+![图 12](https://cdn-images-1.medium.com/max/2000/1*fNpJ18MfE32UD0ib3LnrfA.png)
 
-LSTMs are amazing!
+LSTMs 太令人惊奇了！
```
Comment (TrWestdoor) [inline TODO1/time-series-analysis-visualization-forecasting-with-lstm.md]:
```suggestion
LSTMs 太神奇了！
```
个人觉得此处简单的口语化更合适一点。

**Comment by TrWestdoor** [comment]
@Minghao23 @fanyijihua 校对完成

File: TODO1/time-series-analysis-visualization-forecasting-with-lstm.md (hunk 6 above)
Comment (Minghao23) [reply TODO1/time-series-analysis-visualization-forecasting-with-lstm.md]: 这个我个人觉得不需要改，就先不改了。

**Comment by Minghao23** [comment]
@leviding 修改完成

**Comment by leviding** [comment]
@Minghao23 已经 merge 啦~ 快快麻溜发布到[掘金](https://juejin.im)然后给我发下链接，方便及时添加积分哟。

文章发布时，标题前面那个大图，你可以在上传掘金的时候作为题图使用，然后在文中删除那个图就行啦。

掘金翻译计划有自己的知乎专栏，你也可以投稿哈，推荐使用一个[好用的插件](https://chrome.google.com/webstore/detail/minimalist-markdown-edito/pghodfjepegmciihfhdipmimghiakcjf?hl=zh-CN)。
专栏地址：https://zhuanlan.zhihu.com/juejinfanyi

**Comment by Minghao23** [comment]
https://juejin.im/post/5cecdbb75188252db706f4e9
谢谢~
